lib/spack/spack/fetch_strategy.py

```diff
@@ -90,17 +90,23 @@ class FetchStrategy(with_metaclass(FSMeta, object)):
     #: classes have multiple ``url_attrs`` at the top-level.
     optional_attrs = []  # optional attributes in version() args.
 
-    def __init__(self):
+    def __init__(self, **kwargs):
         # The stage is initialized late, so that fetch strategies can be
         # constructed at package construction time.  This is where things
         # will be fetched.
         self.stage = None
+        # Possibly disable caching for this strategy.
+        self._no_cache_opt = kwargs.pop('no_cache', False)
```

Comment:
Is there a reason you want to remove `no_cache` from `kwargs` using `pop` (versus `get`)?

Reply:
I popped it from `kwargs` since the option has now been dealt with and does not need to be in `kwargs` as seen by sub-classes: just separation of concerns, basically.

Per your second point: I have a natural inclination to avoid options that are true by default. I could certainly see the attribute being something like `self.cache_enabled`, but I'd like to continue to have the option name reflect what happens when it is set to the non-default `True` case. Does this make sense to you?

Reply:
I can see your reasoning in both cases and neither point is a blocker.

I am curious about the basis for your natural inclination simply for my own edification.

Reply:
I'm not sure I would claim my natural inclination had a conscious, "basis"—or was even entirely rational—but I guess that when I'm adding an option to something, I like specifying the option to be an affirmative act _i.e._ I'm specifying the option explicitly because I want to _change_ something about how it (function, command, _etc._) behaves. I would therefore like the name of the option to reflect what I want to happen when I specify it, which (in my mind) equates to wanting the option to be `True` when I specify it.

I hope that makes sense to you. It's a bit of a _post hoc_ rationalization since I can't actually put my finger on exactly when or why I started feeling this way.

Reply:
I can appreciate your perspective.  Thanks for sharing.